### ISSUE TITLE
Unexpected unnecessary_chaining warning in types

### ISSUE THREAD

**Issue by sbcgua** [abaplint/abaplint#3612]
```abap
  types: " unnecessary_chaining raise here
    begin of Customer.
      include type CompanyStructure.
      types:
      " Extension fields
      CustomerID type SAFmiddle1textType,
      CustomerType type SAFmiddle1textType,
      RelatedPartyCode type SAFcodeType,
      CustomerBalance type CounterAgentBalances,
    end of Customer.
```

unnecessary_chaining raises on type definition
